test/Faker/Provider/nl_NL/PersonTest.php

```diff
@@ -21,6 +21,71 @@ public function testGenerateValidIdNumber()
         self::assertTrue($sum != 0 && $sum % 11 == 0);
     }
 
+    /**
+     * @dataProvider identityDocumentNumberProvider
+     * @param \DateTime   $issueDate
+     * @param string|null $documentCode
+     * @param string      $regex
+     */
+    public function testGenerateValidIdentityDocumentNumber(\DateTime $issueDate, ?string $documentCode, string $regex)
+    {
+        self::assertMatchesRegularExpression($regex, $this->faker->identityDocumentNumber($issueDate, $documentCode));
+    }
+
+
+    /**
+     * @dataProvider validityProvider
+     * @param int $age
+     * @param int $validity
+     */
+    public function testGenerateValidIdentityDocumentValidity(int $age, int $validity)
+    {
+        self::assertSame($validity, $this->faker->identityDocumentValidity($age));
+    }
+
+    /**
+     * @return array
+     */
+    public function identityDocumentNumberProvider()
+    {
+        return [
+            [new \DateTime('2019-11-30'), null, '/^[INDSARB]((?![O])[A-Z])((?![O])[A-Z0-9]){6}[0-9]{1}$/'],
```

Comment:
Reading through it again, as it's implemented now  the second number can only be 1-9, so this actually should be 
`/^[INDSARB]((?![O])[A-Z])((?![O])[A-Z0-9]){6}[1-9]{1}$/`

By the  way, I couldn't find a source that mentions the December 1st 2019 special case - could  you add a link to the docblock?
The only thing I could find is variations of this: https://www.fxglobe.com/wp-content/uploads/2018/11/esma70.pdf where it is actually mentioned that the last number may indeed be 0-9 (not 1-9 as it is implemented here currently).

Reply:
- Fixed the generator (test was correct).
- Info from https://www.rvig.nl/reisdocumenten/vraag-en-antwoord/hoe-is-het-paspoortnummer-of-identiteitskaartnummer-opgebouwd (Dutch) which translates to:
`Since December 1, 2019, the figure "0" is no longer mentioned in newly issued passports and Dutch Identity Cards.`

Reply:
Great, thanks!